src/components/App.js

```diff
@@ -1,17 +1,222 @@
+        this.searchByID = this.searchByID.bind(this);
+        this.toggleVisible = this.toggleVisible.bind(this);
+        this.toggleMinify = this.toggleMinify.bind(this);
+    }
+
+    componentDidMount() {
+        if (window.localStorage.favourites != undefined) {
+            this.setState({
+                favourites: JSON.parse(window.localStorage.getItem("favourites"))
+            })
+        }
+    }
+
+    searchByTitle(query, pageNum = 1) {
+        fetch(`http://www.omdbapi.com/?apikey=507b4100&type=movie&s=${query}&page=${pageNum}`)
+            .then(response => response.json())
+            .then(body => {
+                this.setState({
+                    hints: [],
+                    films: body.Search,
+                    totalFilms: body.totalResults
+                });
+            })
+            .catch(error => {
+                alert(error);
+            });
+    }
+
+    searchByID(id) {
+        fetch(`http://www.omdbapi.com/?apikey=507b4100&type=movie&i=${id}`)
+            .then(response => response.json())
+            .then(body => {
+                this.setState({
+                    filmDetails: body,
+                    hints: []
+                }, () => this.toggleVisible())
+            })
+            .catch(error => {
+                alert(error);
+            })
+    }
+
+    receiveTitleQuery(query) {
+        this.setState({
+            query: query,
+            currentPage: 1,
+            filmDetails: {}
+        }, () => this.searchByTitle(query, this.state.currentPage));
+    }
+
+    receiveSearchHint(string) {
+        this.setState({
+            query: string,
+        }, () => this.searchHint(this.state.query));
+    }
+
+    searchHint(string, pageNum = 1) {
+        fetch(`http://www.omdbapi.com/?apikey=507b4100&type=movie&s=${string}&page=${pageNum}`)
+            .then(response => response.json())
+            .then(body => {
+                this.setState({
+                    hints: body.Search
+                })
+            })
+            .catch(error => {
+                alert(error);
+            })
+    }
+
+    receiveFilmID(id) {
+        this.searchByID(id);
+    }
+
+    receivePageNum(num) {
+        this.setState({
+            currentPage: num
+        }, () => this.searchByTitle(this.state.query, this.state.currentPage));
+    }
+
+    receiveFavListState(boolean) {
+        this.setState({
+            showFavs: boolean
+        });
+    }
+
+    // TODO: refactor using concat() and filter() to avoid mutation of favourites array
+
+    receiveFav(obj) {
+        let favourites = this.state.favourites;
+        const isFav = film => {
+            return film.imdbID == obj.imdbID
+        }
+        if (!favourites.find(isFav)) {
+            favourites.unshift(obj);
+            this.setState({
+                favourites: favourites
+            }, () => window.localStorage.setItem("favourites", JSON.stringify(this.state.favourites)));
```

Comment:
I would take `window.localStorage.setItem("favourites", JSON.stringify(this.state.favourites))` out of callback and use `window.localStorage.setItem("favourites", JSON.stringify(favourites))`